modules/spark/src/main/scala/notebook/spark/util/SparkClosureUtil.scala

```diff
@@ -0,0 +1,59 @@
+package notebook.spark.util
+
+import scala.reflect.runtime.universe
+import org.apache.log4j.AppenderSkeleton
+import org.apache.log4j.spi.LoggingEvent
+
+class SparkClosureUtil(appendTo: String => Unit = println) {
+
+  // get the closure cleaner
+  private val runtimeMirror = universe.runtimeMirror(getClass.getClassLoader)
+  private val module = runtimeMirror.staticModule("org.apache.spark.util.ClosureCleaner")
```

Comment:
well, you don't have to make the classes of`org.apache.spark.spark_notebook_hacks` private, and it will be able to acess the spark stuff. we've used this before...